### PR TITLE
places: new resource bf:Place

### PR DESCRIPTION
**Pull request by rerowep**
* Adds place.
* Removes agents redirects.

### CODE REVIEW

File: rero_mef/places/idref/tasks.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        days_spann=30,
+        from_date=from_date,
+        until_date=until_date,
+        verbose=verbose
+    )
+
+
+@shared_task
+def idref_get_record(id, debug=False):
+    """Get a record from GND OAI repo."""
```
Comment (PascalRepond) [inline rero_mef/places/idref/tasks.py:119]: Docstring incorrect "GND" should be "IDREF"

File: rero_mef/marctojson/do_idref_places.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            self.json_dict['type'] = 'bf:Place'
+
+    def trans_idref_bnf_type(self):
+        """Transformation bnf_type from field 008."""
+        if self.logger and self.verbose:
+            self.logger.info('Call Function', 'trans_idref_bnf_type')
+        if field_008 := self.marc.get_fields('008'):
+            self.json_dict['bnf_type'] = 'sujet Rameau'
+            if field_008[0].data == 'Tf8':
+                self.json_dict['bnf_type'] = 'genre/forme Rameau'
```
Comment (PascalRepond) [inline rero_mef/marctojson/do_idref_places.py:72]: I think this function is not needed for places. Places cannot be genre/forme Rameau, I think they will always be "sujet Rameau".

File: rero_mef/places/idref/mappings/v7/places_idref/idref-place-v0.0.1.json [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            "type": "keyword"
+          },
+          "value": {
+            "type": "keyword"
+          }
+        }
+      },
+      "bnf_type": {
+        "type": "text"
+      },
```
Comment (PascalRepond) [inline rero_mef/places/idref/mappings/v7/places_idref/idref-place-v0.0.1.json:39]: Do we really need/use this field? It is useful for concepts to differentiate subjects from genre/form but places will always be subjects

Comment (rerowep) [reply rero_mef/places/idref/mappings/v7/places_idref/idref-place-v0.0.1.json:39]: The structure is more or less like in concepts, thats way I have not changed this.

**Review by PascalRepond — APPROVED** [review]

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+      "title": "Place record ID",
+      "type": "string",
+      "minLength": 1
+    },
+    "type": {
+      "title": "Type",
+      "type": "string",
+      "enum": [
+        "bf:Place"
+      ]
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:29]: It seems to be a constant ? why not use `"const"` jsonschema attribute : 

```json
"type": {
    "title": "Type",
    "type": "string",
    "readOnly": true,
    "default": "bf:Place",
    "const": "bf:Place"
}
```

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+          },
+          "source": {
+            "enum": [
+              "BNF",
+              "RERO",
+              "IDREF"
+            ]
+          },
+          "value": {
+            "type": "string"
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:61]: add a minLength constraint

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+      "title": "Authorized access point",
+      "type": "string",
+      "minLength": 1
+    },
+    "variant_access_point": {
+      "title": "Variant access point",
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "string"
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:81]: minLength ?

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "properties": {
+          "type": {
+            "enum": [
+              "bf:ClassificationDdc"
+            ]
+          },
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:170]:
```json
"type": {
    "type": "string",
    "const": "bf:ClassificiationDdc"
}
```

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+              "seeReference",
+              "seeAlsoReference",
+              "REROtreatment"
+            ]
+          },
+          "label": {
+            "type": "array",
+            "minItems": 1,
+            "items": {
+              "type": "string"
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:205]: minLength ?

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+                "enum": [
+                  "bf:Local",
+                  "uri"
+                ]
+              },
+              "source": {
+                "type": "string"
+              },
+              "value": {
+                "type": "string"
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:239]: minLength

File: rero_mef/jsonschemas/common/place-v0.0.1.json [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+              }
+            }
+          }
+        }
+      }
+    },
+    "deleted": {
+      "title": "Deletion date",
+      "type": "string"
```
Comment (zannkukai) [inline rero_mef/jsonschemas/common/place-v0.0.1.json:248]: add `"format": "date-time"`?

File: rero_mef/monitoring/api.py [hunk 11]
```diff
@@ -95,7 +95,7 @@ def get_es_count(cls, index):
         """
         try:
             result = RecordsSearch(index=index).query().count()
-        except NotFoundError:
+        except NotFoundError as err:
```
Comment (zannkukai) [inline rero_mef/monitoring/api.py:98]: changes not required ; you never use `err`

File: rero_mef/places/idref/tasks.py [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+from flask import current_app
+from sickle import OAIResponse, Sickle
+
+from .api import PlaceIdrefRecord
+from ...marctojson.do_idref_places import Transformation
+from ...utils import MyOAIItemIterator, oai_get_record, \
+    oai_process_records_from_dates, oai_save_records_from_dates
+
+
+class MySickle(Sickle):
```
Comment (zannkukai) [inline rero_mef/places/idref/tasks.py:32]: `MySickle` seems not a very useful name

File: rero_mef/places/mef/jsonschemas/places_mef/mef-place-v0.0.1.json [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+      "title": "Mef place ID",
+      "type": "string",
+      "minLength": 1
+    },
+    "type": {
+      "title": "Type",
+      "type": "string",
+      "enum": [
+        "bf:Place"
+      ]
```
Comment (zannkukai) [inline rero_mef/places/mef/jsonschemas/places_mef/mef-place-v0.0.1.json:28]: const ?

File: rero_mef/places/mef/jsonschemas/places_mef/mef-place-v0.0.1.json [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        "$ref": {
+          "title": "ref",
+          "type": "string",
+          "pattern": "^https://mef.rero.ch/api/places/idref/.*?$"
+        }
+      }
+    },
+    "deleted": {
+      "title": "Deletion date",
+      "type": "string"
```
Comment (zannkukai) [inline rero_mef/places/mef/jsonschemas/places_mef/mef-place-v0.0.1.json:43]: format ?

File: rero_mef/concepts/idref/api.py [hunk 15]
```diff
@@ -52,6 +52,15 @@ class ConceptIdrefRecord(ConceptRecord):
     model_cls = ConceptIdrefMetadata
     search = ConceptIdrefSearch
 
+    @classmethod
+    def get_online_record(cls, id, debug=False):
+        """Get online Record.
+
+        Has to be overloaded in agent class.
+        """
+        from .tasks import idref_get_record
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_mef/concepts/idref/api.py:61]: It is in tasks because we can call it asynchron.

File: rero_mef/places/idref/tasks.py (hunk 12 above)
Comment (rerowep) [reply rero_mef/places/idref/tasks.py:32]: Renamed and moved to utils.

**Pull request merged**